lux-lib/src/config/mod.rs

```diff
@@ -603,6 +620,109 @@ impl UserData for Config {
                 .map_err(|err| err.into_lua_err())
         });
 
-        // TODO: Implement builder with `require("lux").config.build()`
+        methods.add_function("builder", |_, ()| ConfigBuilder::new().into_lua_err());
+
+        methods.add_method("server", |_, this, ()| Ok(this.server().to_string()));
+        methods.add_method("extra_servers", |_, this, ()| {
+            Ok(this
+                .extra_servers()
+                .iter()
+                .map(|url| url.to_string())
+                .collect_vec())
+        });
+        methods.add_method("only_sources", |_, this, ()| {
+            Ok(this.only_sources().cloned())
+        });
+        methods.add_method("namespace", |_, this, ()| Ok(this.namespace().cloned()));
+        methods.add_method("lua_dir", |_, this, ()| Ok(this.lua_dir().cloned()));
+        methods.add_method("lua_version", |_, this, ()| Ok(this.lua_version().cloned()));
+        methods.add_method("tree", |_, this, lua_version: LuaVersion| {
+            this.tree(lua_version).into_lua_err()
+        });
+        methods.add_method("luarocks_tree", |_, this, ()| {
+            Ok(this.luarocks_tree().clone())
+        });
+        methods.add_method("no_project", |_, this, ()| Ok(this.no_project()));
+        methods.add_method("verbose", |_, this, ()| Ok(this.verbose()));
+        methods.add_method("timeout", |_, this, ()| Ok(this.timeout().as_secs()));
+        methods.add_method("cache_dir", |_, this, ()| Ok(this.cache_dir().clone()));
+        methods.add_method("data_dir", |_, this, ()| Ok(this.data_dir().clone()));
+        methods.add_method("entrypoint_layout", |_, this, ()| {
+            Ok(this.entrypoint_layout().clone())
+        });
+        methods.add_method("variables", |_, this, ()| Ok(this.variables().clone()));
+        // FIXME: This is a temporary workaround to get the external_deps hooked up to Lua
+        // methods.add_method("external_deps", |_, this, ()| {
+        //     Ok(this.external_deps().clone())
+        // });
```

Comment:
question: is this copied over from somewhere else?

Reply:
this is as it should be: it should be implemented but I haven't yet implemented UserData for the `ExternalDependencySearchConfig` struct